File: PL/Scripts/Empleado/Empleado.js
$(document).ready(function () { //click
    GetAll();
    EstadoGetAll();
});

function GetAll() {
    $.ajax({
        type: 'GET',
        url: 'http://localhost:19691/api/Empleado/GetAll',
        success: function (result) { //200 OK
            $('#SubCategorias tbody').empty();
            $.each(result.Objects, function (i, empleado) {
                var filas = '<tr>' +
                    '<td class="text-center"> <button class="btn btn-warning" onclick="GetById(' + empleado.IdEmpleado + ')"><span class="glyphicon glyphicon-edit" style="color:#FFFFFF"></span></button>' + '</td>'
                    + "<td  id='id' class='hidden'>" + empleado.IdEmpleado + "</td>" + "<td class='text-center'>" + empleado.NumeroNomina + "</td>" + "<td class='text-center'>" + empleado.Nombre + "</td>" + "<td class='text-center'>" + empleado.ApellidoPaterno + "</ td>" + "<td class='text-center'>" + empleado.ApellidoMaterno + "</td>" + "<td class='text-center'>" + empleado.Estado.Nombre + "</td>"
                    //+ '<td class="text-center">  <a href="#" onclick="return Eliminar(' + subCategoria.IdSubCategoria + ')">' + '<img  style="height: 25px; width: 25px;" src="../img/delete.png" />' + '</a>    </td>'
                    + '<td class="text-center"> <button class="btn btn-danger" onclick="Eliminar(' + empleado.IdEmpleado + ')"><span class="glyphicon glyphicon-trash" style="color:#FFFFFF"></span></button></td>'

                    + "</tr>";
                $("#SubCategorias tbody").append(filas);
            });
        },
        error: function (result) {
            alert('Error en la consulta.' + result.responseJSON.ErrorMessage);
        }
    });
};

function Add(empleado) {
        $.ajax({
            type: 'POST',
            url: 'http://localhost:19691/api/Empleado/Add',
            dataType: 'json',
            data: empleado,
            success: function (result) {
                $('#myModal').modal();
                $('#ModalUpdate').modal('hide');
                GetAll();
            },
            error: function (result) {
                alert('Error en la consulta.' + result.responseJSON.ErrorMessage);
            }
        });
    };

function GetById(IdEmpleado) {
    $.ajax({
        type: 'GET',
        url: 'http://localhost:19691/api/Empleado/GetById/' + IdEmpleado,
        success: function (result) {
            $('#txtIdEmpleado').val(result.Object.IdEmpleado);
            $('#txtNumeroNomina').val(result.Object.NumeroNomina);
            $('#txtNombre').val(result.Object.Nombre);
            $('#txtApellidoP').val(result.Object.ApellidoPaterno);
            $('#txtApellidoM').val(result.Object.ApellidoMaterno);
            $('#ddlEstado').val(result.Object.Estado.IdEstado);

            $('#ModalUpdate').modal('show');
        },
        error: function (result) {
            alert('Error en la consulta.' + result.responseJSON.ErrorMessage);
        }


    });

}

function EstadoGetAll() {
    $.ajax({
        type: 'GET',
        url: 'http://localhost:19691/api/Estado/GetAll',
        success: function (result) {
            $("#ddlEstado").append('<option value="' + 0 + '">' + 'Seleccione una opción' + '</option>');
            $.each(result.Objects, function (i, estado) {
                $("#ddlEstado").append('<option value="'
                    + estado.IdEstado + '">'
                    + estado.Nombre + '</option>');
            });
        }
    });
}

function Update(empleadoUD) {
    
    $.ajax({
        type: 'POST',
        url: 'http://localhost:19691/api/Empleado/Update',
        datatype: 'json',
        data: empleadoUD,
        success: function (result) {
            $('#myModal').modal();
            $('#ModalUpdate').modal('hide');
            GetAll();
        },
        error: function (result) {
            alert('Error en la consulta.' + result.responseJSON.ErrorMessage);
        }
    });
};

function Modal() {
    var mostrar = $('#ModalUpdate').modal('show');
    IniciarEmpleado();

}

function Actualizar() {
    var empleado1 = {
        IdEmpleado: $('#txtIdEmpleado').val()
    }

    if (empleado1.IdEmpleado == '') {
        var empleado = {
            IdEmpleado: 0,
            NumeroNomina: $('#txtNumeroNomina').val(),
            Nombre: $('#txtNombre').val(),
            ApellidoPaterno: $('#txtApellidoP').val(),
            ApellidoMaterno: $('#txtApellidoM').val(),
            Estado: {
                IdEstado: $('#ddlEstado').val()
            }
        }
        Add(empleado);

    }
    else {
        var empleadoUD = {
            IdEmpleado: $('#txtIdEmpleado').val(),
            NumeroNomina: $('#txtNumeroNomina').val(),
            Nombre: $('#txtNombre').val(),
            ApellidoPaterno: $('#txtApellidoP').val(),
            ApellidoMaterno: $('#txtApellidoM').val(),
            Estado: {
                IdEstado: $('#ddlEstado').val()
            }
        }
        Update(empleadoUD);
    }
}

function Eliminar(IdEmpleado) {
    if (confirm("¿Estas seguro de eliminar el empleado ?")) {
        $.ajax({
            type: 'GET',
            url: 'http://localhost:19691/api/Empleado/Delete/' + IdEmpleado,
            success: function (result) {
                $('#myModal').modal();
                GetAll();
            },
            error: function (result) {
                alert('Error en la consulta' + result.responseJSON.ErrorMessage);
            }
        });

    };
};


function IniciarEmpleado() {

    var empleado = {
        IdEmpleado: $('#txtIdEmpleado').val(''),
        NumeroNomina: $('#txtNumeroNomina').val(''),
        Nombre: $('#txtNombre').val(''),
        ApellidoPaterno: $('#txtApellidoP').val(''),
        ApellidoMaterno: $('#txtApellidoM').val(''),
        Estado: {
            IdEstado: $('#ddlEstado').val(0)
        }
    }
}

File: BL/Empleado.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BL
{
    public class Empleado
    {
        public static ML.Result Add(ML.Empleado empleado)
        {
            ML.Result result = new ML.Result();
            try
            {
                using (DL.ExamenJavaScriptEntities context = new DL.ExamenJavaScriptEntities())
                {
                    var query = context.EmpleadoAdd(empleado.NumeroNomina,empleado.Nombre,empleado.ApellidoPaterno,empleado.ApellidoMaterno,empleado.Estado.IdEstado);

                    if (query >= 1)
                    {
                        result.Correct = true;
                    }
                    else
                    {
                        result.Correct = false;
                        result.ErrorMessage = "No se ha podido realizar el insert";
                    }
                    result.Correct = true;
                }
            }

            catch (Exception ex)
            {
                result.Correct = false;
                result.ErrorMessage = ex.Message;
            }
            return result;
        }
        public static ML.Result Update(ML.Empleado empleado)
        {
            ML.Result result = new ML.Result();
            try
            {
                using (DL.ExamenJavaScriptEntities context = new DL.ExamenJavaScriptEntities())
                {
                    var query = context.EmpleadoUpdate(empleado.IdEmpleado,empleado.NumeroNomina, empleado.Nombre, empleado.ApellidoPaterno, empleado.ApellidoMaterno, empleado.Estado.IdEstado);

                    if (query >= 1)
                    {
                        result.Correct = true;
                    }
                    else
                    {
                        result.Correct = false;
                        result.ErrorMessage = "No se ha podido realizar el update";
                    }
                    result.Correct = true;
                }
            }

            catch (Exception ex)
            {
                result.Correct = false;
                result.ErrorMessage = ex.Message;
            }
            return result;
        }
        public static ML.Result Delete(ML.Empleado empleado)
        {
            ML.Result result = new ML.Result();
            try
            {
                using (DL.ExamenJavaScriptEntities context = new DL.ExamenJavaScriptEntities())
                {
                    var query = context.EmpleadoDelete(empleado.IdEmpleado);

                    if (query >= 1)
                    {
                        result.Correct = true;
                    }
                    else
                    {
                        result.Correct = false;
                        result.ErrorMessage = "No se ha podido realizar el insert";
                    }
                    result.Correct = true;
                }
            }

            catch (Exception ex)
            {
                result.Correct = false;
                result.ErrorMessage = ex.Message;
            }
            return result;
        }
        public static ML.Result GetById(int IdEmpleado)
        {
            ML.Result result = new ML.Result();
            try
            {
                using (DL.ExamenJavaScriptEntities context = new DL.ExamenJavaScriptEntities())
                {
                    var obj = context.EmpleadoGetById(IdEmpleado).FirstOrDefault();
                    result.Objects = new List<object>();

                    if (obj != null)
                    {
                        ML.Empleado empleado = new ML.Empleado();
                        empleado.IdEmpleado = obj.IdEmpleado;
                        empleado.NumeroNomina = obj.NumeroNomina;
                        empleado.Nombre = obj.Nombre;
                        empleado.ApellidoPaterno = obj.ApellidoPaterno;
                        empleado.ApellidoMaterno = obj.ApellidoMaterno;

                        empleado.Estado = new ML.Estado();
                        empleado.Estado.IdEstado = obj.IdEstado;
                        empleado.Estado.Nombre = obj.Estado;

                        result.Object = empleado;
                        result.Correct = true;

                    }
                    else
                    {
                        result.Correct = false;
                        result.ErrorMessage = "No se pudo realizar la consulta";
                    }

                }
            }

            catch (Exception ex)
            {
                result.Correct = false;
                result.ErrorMessage = ex.Message;
                result.Ex = ex;
            }
            return result;
        }
        public static ML.Result GetAll()
        {
            ML.Result result = new ML.Result();
            try
            {
                using (DL.ExamenJavaScriptEntities context = new DL.ExamenJavaScriptEntities())
                {
                    var empleados = context.EmpleadoGetAll().ToList();
                    result.Objects = new List<object>();
                    if (empleados != null)
                    {
                        foreach (var obj in empleados)
                        {
                            ML.Empleado empleado = new ML.Empleado();
                            empleado.IdEmpleado = obj.IdEmpleado;
                            empleado.NumeroNomina = obj.NumeroNomina;
                            empleado.Nombre = obj.Nombre;
                            empleado.ApellidoPaterno = obj.ApellidoPaterno;
                            empleado.ApellidoMaterno = obj.ApellidoMaterno;

                            empleado.Estado = new ML.Estado();
                            empleado.Estado.IdEstado = obj.IdEstado;
                            empleado.Estado.Nombre = obj.Estado;

                            result.Objects.Add(empleado);

                        }
                        result.Correct = true;
                    }
                    else
                    {
                        result.Correct = false;
                        result.ErrorMessage = "No se ha podido realizar la consulta";

                    }
                }
            }
            catch (Exception ex)
            {
                result.Correct = false;
                result.ErrorMessage = ex.Message;
            }
            return result;
        }
    }
}
